rsocket-core/src/main/java/io/rsocket/core/CleanOnClearQueueDecorator.java

```diff
@@ -0,0 +1,108 @@
+package io.rsocket.core;
+
+import io.netty.util.ReferenceCountUtil;
+import io.rsocket.Payload;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Queue;
+
+final class CleanOnClearQueueDecorator implements Queue<Payload> {
+  final Queue<Payload> delegate;
+
+  CleanOnClearQueueDecorator(Queue<Payload> delegate) {
+    this.delegate = delegate;
+  }
+
+  @Override
+  public void clear() {
+    Payload p;
+    while ((p = delegate.poll()) != null) {
+      ReferenceCountUtil.safeRelease(p);
+    }
+  }
```

Comment:
So this will clear anything remaining in the `UnicastProcessor` queue after cancellation. What about items that may be dropped, depending on exact timing, before they make it into the queue? Shouldn't we also have an `onNextDropped` hook installed as another part of a workaround? An application could also do it but they shouldn't have to.

Reply:
I decided to simplify the mechanism. This queue directly specified for `Payload`s. Now, this queue extends AtomicBoolean so once clear called (and it is called only on terminal signal) we set the volatile field to `true` so all  the late Payloads coming into the queue will be immediately released

Reply:
What I can do as well is hacking publishOn operator over reflection and provide cleanOnClearQueueDecorator for the default queue to a contractor to ensure that we can workaround racing-related leaks for now. What do you think about that @rstoyanchev?

Reply:
That's covering even more places, good.